### DISCUSSION TITLE
What the meaning of some descriptors? (HallKierAlpha and Kappa1-3)

### DISCUSSION THREAD

**Discussion by BeMgCaSrBa** [rdkit/rdkit#6884] [Q&A] [answered]
What's the meaning of HallKierAlpha and Kappa1-3? I can't find enough information about these descriptors. Where can I learn more about them?

**Comment by dehaenw** [comment, marked as answer]
These are topological indices. These specific topological indices are said to contain some kind of shape information and make use of atom radii, and topological path counts.
You may find their reference here:
https://www.rdkit.org/docs/cppapi/namespaceRDKit_1_1Descriptors.html
eqns (58-62) of [Rev. Comp. Chem. vol 2, 367-422, (1991)](https://onlinelibrary.wiley.com/doi/epdf/10.1002/9780470125793.ch9)

$\alpha_A$ is calculated on an atom per atom basis:
![image](https://github.com/rdkit/rdkit/assets/66372095/cf45d02f-9641-4984-89a9-65891ff0788d)
with r being the covalent radius of the atom and rCsp<sup>3</sup> the covalent radius of a sp<sup>3</sup> hybridized carbon. The paper gives a list of alpha values for various atom types.

Kappa is then calculated as follows (with $\alpha$ the sum of all atoms $\alpha_A$'s):
![image](https://github.com/rdkit/rdkit/assets/66372095/19d59de5-9bcd-4d48-b5a4-f1e42f01ca44)
With A the heavy atom count, <sup>n</sup>P<sub>i</sub> the number of paths with length n.  

It's explained in more detail in that reference, in case you need more precise info.

**Reply by BeMgCaSrBa** [reply]
Thank you for your detailed answer！

**Reply by BeMgCaSrBa** [reply]
Hello, I have read this reference carefully, I think the formula 62 may be wrong, the square should be in the first term: （A+alpha-2)^2*（A+alpha-3). Because the "3Pmax" is (A-2)^2/4 when A is envn. Hopefully this issue didn't affect the code, or my own understanding is faulty.

**Reply by dehaenw** [reply]
I just had a look and indeed when reasoning it out, it turns out eq. 55 and eq. 62 are wrong in the reference, in the way you say. Formula 55 should read: (edit:corrected formulas)

$^3\kappa = {(A -3) (A-2)^2 \over (^3P_{i})^2}$

likewise, equation 62 should be:

$^3\kappa_{\alpha} = {(A+\alpha -3) (A+\alpha-2)^2 \over (^3P_{i}+\alpha)^2}$

I believe the docs state the formula is implemented as stated in the reference, so I am quite sure it will be the uncorrected one.

**Reply by BeMgCaSrBa** [reply]
Will this be fixed in a future release? Or is it going to stay that way? I feel like if it is fixed, the impact might be even greater than if not.

**Reply by dehaenw** [reply]
That's a good question, but one for the maintainers, such as @greglandrum , to consider. It's fair to say changing it now might cause more problems than it fixes.

BTW,
for a test case to see the difference, check out this codeblock that calculates alpha adjusted 3Kappa both according to rdkit descriptors and according to the wrong and corrected formula. This confirms that the formula is used as provided, not the corrected one. Any molecule with an even amount of atom will do.

```python 
from rdkit import Chem
from rdkit.Chem import Descriptors
m=Chem.MolFromSmiles("c1ccoc1CCN")
print("RDKit 3Kappa:",Descriptors.Kappa3(m))
A=len(m.GetAtoms())
Pi=len(Chem.FindAllPathsOfLengthN(m,3))
alpha=Descriptors.HallKierAlpha(m)
if A%2==0:
    print("correct 3Kappa:",(A+alpha-3)*(A+alpha-2)**2/((Pi+alpha)**2))
    print("wrong 3Kappa:",(A+alpha-3)**2*(A+alpha-2)/((Pi+alpha)**2))
else:
    print("correct 3Kappa:",(A+alpha-1)*(A+alpha-3)**2/((Pi+alpha)**2))
```
this prints out 
```
RDKit 3Kappa: 1.103364629598396
correct 3Kappa: 1.3635921365791497
wrong 3Kappa: 1.103364629598396
```
